app/factories/flood_risk_engine/steps/form_object_factory.rb

```diff
@@ -38,11 +38,16 @@ def form_object_class_map
         private
 
         def setup_form_object(step)
-          "FloodRiskEngine::Steps::#{step.to_s.classify}Form".constantize
-        rescue
-          Rails.logger.debug(" No Form Object found for step [#{step}] - falling back to object_class map")
-          nil
+          form_name = "FloodRiskEngine::Steps::#{step.to_s.classify}Form"
+          begin
+            form_name.constantize
+          rescue => x
```

Comment:
If we are going down the meta route for determining the form name, wouldn't it be easier to check `form_object_class_map` first and then try to work out the form class from the step name if the step isn't in the list? Then we wouldn't need this rescue clause.

Then as we add named forms we can remove the respective step from `form_object_class_map`. Ultimately, `form_object_class_map` will empty and at that point we can drop it.

If the exception catching is to stay then it should be limited to only catch the error generated because the constant doesn't exist. That is:

```
  rescue NameError => x
```